lib/routes/cline/blog.ts

```diff
@@ -7,56 +7,110 @@ const rootUrl = 'https://cline.bot';
 const blogUrl = `${rootUrl}/blog`;
 
 // Extract article information from DOM
-function extractArticlesFromDOM($) {
-    return $('article')
-        .toArray()
-        .map((article) => {
-            const element = $(article);
-            const title = element.find('h2').first().text().trim();
-            const linkEl = element.find('a').first();
-            const link = linkEl.attr('href');
-            const fullLink = link ? (link.startsWith('http') ? link : `${rootUrl}${link.startsWith('/') ? link : `/${link}`}`) : '';
+function extractArticlesFromDOM($: CheerioAPI): DataItem[] {
+    const articles: DataItem[] = [];
 
-            const metaEl = element.find('.text-xs.text-slate-500');
-            const author = metaEl.find('span').first().text().trim();
-            const dateStr = metaEl.find('span').eq(2).text().trim();
-            const pubDate = dateStr ? parseDate(dateStr) : undefined;
+    // Archive page structure - articles in chronological list
+    $('div.group > div.space-y-4 > div.group').each((_, article) => {
+        const element = $(article);
 
-            const summary = element.find('.text-slate-600').text().trim();
-            const imgSrc = element.find('img').attr('src') || '';
+        // Title and link
+        const titleEl = element.find('h3.text-lg.font-semibold.text-gray-900 a, a.text-gray-900');
+        const title = titleEl.text().trim();
+        const link = titleEl.attr('href') || element.find('a[href*="/blog/"]').attr('href');
+        const fullLink = link ? (link.startsWith('http') ? link : `${rootUrl}${link.startsWith('/') ? link : `/${link}`}`) : '';
 
-            return {
+        // Date extraction from format like "Aug 12, 2024"
+        const dateEl = element.find('div.text-sm.text-gray-500');
+        const dateStr = dateEl.text().trim();
+        const pubDate = dateStr ? parseDate(dateStr) : undefined;
+
+        // Description
+        const summary = element.find('p.text-gray-600').text().trim();
+
+        if (title && link) {
+            articles.push({
```

Comment:
Use [map](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Array/map) instead of [push](https://www.measurethat.net/Benchmarks/Show/11532/0/push-vs-map)